packages/frontend/src/common/components/TextInput.tsx

```diff
@@ -0,0 +1,62 @@
+import { ReactComponentElement } from 'react'
+import { FieldError, FieldValues, Path, UseFormRegister } from 'react-hook-form'
+
+interface TextInputProps<T extends FieldValues> {
+  type: string
+  name: Path<T>
+  label: string
+  placeholder: string
+  register: UseFormRegister<T>
+  icon?: ReactComponentElement<any>
+  error?: FieldError
+  additionalClasses?: string
+  additionalLabelClasses?: string
+  additionalInputClasses?: string
+}
+
+export function TextInput<T extends FieldValues>({
+  type,
+  name,
+  label,
+  placeholder,
+  register,
+  icon,
+  error,
+  additionalClasses,
+  additionalLabelClasses,
+  additionalInputClasses,
+}: TextInputProps<T>) {
+  return (
+    <div
+      className={`form-control w-full text-gray-400 ${
+        error?.message && 'text-error'
+      } ${additionalClasses}`}
+    >
+      <label className={`label p-0 ml-4`}>
+        <span
+          className={`label-text leading-6 ${
+            error?.message && 'text-error'
+          } ${additionalLabelClasses}`}
+        >
+          {label}
+        </span>
+      </label>
+      <div className="relative focus-within:text-gray-600">
+        {icon}
+        <input
+          type={type}
+          placeholder={placeholder}
+          className={`input w-full text-sm bg-lightHover text-gray-600 leading-6 placeholder:font-normal focus-within:placeholder:font-normal font-bold ${
+            error?.message && 'input-error bg-error bg-opacity-20'
+          } ${icon && 'pl-12'} ${additionalInputClasses} `}
+          {...register(name)}
+        />
+      </div>
+      {error?.message && (
+        <p className="text-error text-left ml-4 text-sm leading-6">
+          {error?.message}
+        </p>
+      )}
+    </div>
+  )
+}
```

Comment:
Maybe it's because of Tailwind, but I quite don't like how `TextInput` and `TextArea` components are structured 🤔 It seems kind of messy.

I wonder if this can be helpful https://www.tailwind-variants.org/docs/introduction